File: frontend/mobile/lib/services/connections/authenticate.dart
import 'package:area_mobile/services/dio.dart';
import 'package:area_mobile/types/services.dart';

Future<ServiceReturn<String>> authenticate(String serviceId) async {
  try {
    final response = await dio.get('/connections/$serviceId/authenticate');
    return ServiceReturn(data: response.data["oauthUrl"]);
  } catch (e) {
    return ServiceReturn(error: e.toString());
  }
}

File: frontend/web/services/connections/authenticate.ts
import { ServiceReturn } from "@/types/api";
import { axiosInstance } from "@/services";
import { SERVICE_ERROR_UNKNOWN } from "@/config/services";

type AuthenticateResponse = {
	oauthUrl: string;
};
const authenticate = async (serviceId: string): Promise<ServiceReturn<string>> => {
	try {
		const response = await axiosInstance.get<AuthenticateResponse>(`/connections/${serviceId}/authenticate`);
		return { data: response.data.oauthUrl, error: undefined };
	} catch (error) {
		return { data: null, error: SERVICE_ERROR_UNKNOWN };
	}
};

export default authenticate;
